src/bin/directory-cli.rs

```diff
@@ -22,30 +24,32 @@ enum Commands {
     ListAddresses,
 }
 
-fn send_rpc_req(req: &RpcMsgReq) {
-    let mut stream = TcpStream::connect("127.0.0.1:4321").unwrap();
+fn send_rpc_req(req: &RpcMsgReq) -> Result<(), AppError> {
+    let mut stream = TcpStream::connect("127.0.0.1:4321").map_err(MakerError::IO)?;
     stream
         .set_read_timeout(Some(Duration::from_secs(20)))
-        .unwrap();
+        .map_err(MakerError::IO)?;
     stream
         .set_write_timeout(Some(Duration::from_secs(20)))
-        .unwrap();
+        .map_err(MakerError::IO)?;
 
-    send_message(&mut stream, &req).unwrap();
+    send_message(&mut stream, &req).map_err(MakerError::Net)?;
 
-    let resp_bytes = read_message(&mut stream).unwrap();
-    let resp: RpcMsgResp = serde_cbor::from_slice(&resp_bytes).unwrap();
+    let resp_bytes = read_message(&mut stream).map_err(MakerError::Net)?;
+    let resp: RpcMsgResp = serde_cbor::from_slice(&resp_bytes).map_err(MakerError::Deserialize)?;
```

Comment:
Just like MakerError if we add the NetError and ProtocolError to DirectoryError, all these error can be wrapped in DirectoryError itself, instead of Maker Error.

The DirectoryError work is being done in #247, we should include all these changes in this PR itself, and close that PR.